packages/validator/test/acceptance/validate.decorator.acceptance.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  RestApplication,
+  get,
+  param,
+  RestServer,
+  post,
+  requestBody,
+} from '@loopback/rest';
+import {validate, validatable} from '../..';
+import {supertest, createClientForHandler, expect} from '@loopback/testlab';
+import {model, property} from '@loopback/repository';
+import {getJsonSchema, JsonDefinition} from '@loopback/repository-json-schema';
+
+describe('validate decorator', () => {
+  let app: RestApplication;
+  let server: RestServer;
+  let client: supertest.SuperTest<supertest.Test>;
+
+  before(givenAnApplication);
+  before(givenAServer);
+  before(givenAClient);
+  before(async () => {
+    await app.start();
+  });
+
+  after(async () => {
+    await app.stop();
+  });
+
+  function greaterThan10(num: number) {
+    return num > 10;
+  }
+
+  async function asyncGreaterThan10(num: number) {
+    return (await num) > 10;
+  }
+
+  @model()
+  class TestModel {
+    @property() str: string;
+    @property() num: number;
+  }
+
+  class TestController {
+    @get('/simple')
+    @validatable()
+    simple(
+      @param.query.string('str')
+      @validate({format: 'email'})
+      str: string,
+    ) {}
+
+    @get('/multiple/{num1}')
+    @validatable()
+    multiple(
+      @param.query.string('str')
+      @validate({format: 'email'})
+      str: string,
+      @param.path.number('num1')
+      @validate({
```

Comment:
-1 for introducing another decorator `@validate`. IMO, developers should define validation rules as part of the OpenAPI Spec parameter definition.

```ts
@param.path('num1', {type: 'number', multipleOf: 5});
```

That way the OpenAPI spec of the application include all additional constraints imposed on method arguments.

If there are use cases for using `@validate` outside of REST/OpenAPI context, then I am fine with having `@validate` decorator, as long as `@param` decorators invoke the `@validate` decorator under the hood.

Reply:
I agree with you. The PoC was to show that we can use `@validate` as a base for `openapi-v3` decorators, and potentially any other specs like for GraphQL. I didn't have the time to work on the integration itself with `openapi-v3`, but that doesn't seem like something that's difficult to do (call `@validate` on converted spec from `@param`, and call `@validatable` after `@operation`)